crossdock/client/behavior/testify.go

```diff
@@ -0,0 +1,567 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package behavior
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Assert builds an Assertions object that writes success or failure to the
+// given sink.
+//
+// Each assertion made with the Assertions object will result in either a
+// failure or a success being written to the sink.
+func Assert(s Sink) Assertions {
+	return sinkAssertions{s, assert.New(sinkTestingT{s})}
+}
+
+// Require builds an Assertions object that writes success or failure to the
+// given sink. In case of failure, it also stops executing the current
+// behavior.
+//
+// Each assertion made with the Assertions object will result in either a
+// failure or a success being written to the sink.
+func Require(s Sink) Assertions {
+	return sinkAssertions{s, requireAssertions{require.New(sinkTestingT{s})}}
+}
+
+// Fatals builds an Assertions object similar to Require, except that success
+// cases are not written to the Sink, only failures are.
+//
+// Behavior exception stops after the first failure.
+func Fatals(s Sink) Assertions {
+	return requireAssertions{require.New(sinkTestingT{s})}
```

Comment:
basically what's the sink struct function ? we don't have this concept in other language yet.